crates/red_knot/src/types.rs

```diff
@@ -265,6 +284,12 @@ struct ModuleTypeStore {
     symbol_types: FxHashMap<SymbolId, Type>,
     /// cached types of AST nodes in this module
     node_types: FxHashMap<NodeKey, Type>,
+    // the inferred type for symbol K depends on the type of symbols in V
+    symbol_dependencies: FxHashMap<SymbolId, FxHashSet<(FileId, SymbolId)>>,
+    // the inferred type for symbol K depends on the modules in V; this type of dependency is
+    // recorded when e.g. the target symbol doesn't exist in the module, so we can't record a
+    // dependency on a symbol, but if the module changes it could still change our resolution)
```

Comment:
```suggestion
    /// the inferred type for symbol K depends on the modules in V; this type of dependency is
    /// recorded when e.g. the target symbol doesn't exist in the module, so we can't record a
    /// dependency on a symbol, but if the module changes it could still change our resolution)
```